isoprophlex/distributed-postgres
# Execute queries once

This PR closes #TAS-41

## Motivation

We are currently executing twice every query sent to psql: once when we extract it and send it to the sharding crate, where its sent to the backend via rust-postgres, and one more time after that using the already existing implementation.

## Implemented changes

- Change the return type of `SendQueryToShard` in order to return a bool.
- Replace the calls to `SendQuery` in `MainLoop` with calls to `SendQueryToShard`.

## How to test

1. Start a new server, and launch psql.
2. Create a new table, and insert some data.
3. Select the data you inserted previously. 

If the queries are being executed once, the inserts should execute successfully and the select statement should return as many rows as you inserted, with no repeated data.
